vhost-device-gpio/src/backend.rs

```diff
@@ -227,7 +225,7 @@ fn start_backend(args: GpioArgs) -> Result<()> {
     }
 
     for handle in handles {
-        handle.join().map_err(|_| Error::FailedJoiningThreads)??;
+        handle.join().map_err(std::panic::resume_unwind).unwrap()?;
```

Comment:
Not for this PR, but I'm thinking if here (and in i2c and rng) we can have the same issues I had in vsock that I fixed with: https://github.com/rust-vmm/vhost-device/pull/526/commits/8a81f51d00adcbf4d82ec87f7d67d4bed3a353d1

Basically, what happens if a thread other than the first one fails?
The main one is waiting for the first one to finish anyway, so how does it notice the failure?

(I haven't looked at the code, so I don't know if it can happen and mostly if it's a problem)

Reply:
The function documentation on docs.rust-lang.org says "This is designed to be used **in conjunction with** [catch_unwind](https://doc.rust-lang.org/std/panic/fn.catch_unwind.html)" (emphasis mine)

We're not manually catching and resuming unwinding panics here, so it's out of place here.

Reply:
"Any sufficiently complicated concurrent program in another language contains an ad hoc, informally-specified, bug-ridden, slow implementation of half of Erlang."

I agree that this is a problem... I am not sure what the best solution is here... Maybe some RAII-style notifier could be used to cut down the boilerplate code. But let's turn this into a separate issue.

Reply:
@epilys: See the commit log for a reference where this is suggested by the docs.

Reply:
> I agree that this is a problem... I am not sure what the best solution is here... Maybe some RAII-style notifier could be used to cut down the boilerplate code. But let's turn this into a separate issue.

Sure ;-) I was just thinking out loud as I reviewed this code.
I leave this conversation open. @epilys feel free to resolve it about the other topic.

Reply:
Ah, forgot to comment here, but I filed https://github.com/rust-vmm/vhost-device/issues/544.

> I leave this conversation open. @epilys feel free to resolve it about the other topic.

Yep, thats why left this open. Hopefully it was just a misunderstanding :). The thread essentially does the "catch_unwind" part, so we should be good - and do what is recommended.

Reply:
Thanks for linking it in the commit message, github hides them when you're reviewing though :/